### DISCUSSION TITLE
Feature Request: Make FTS index update on INSERT, UPDATE, DELETE

### DISCUSSION THREAD

**Discussion by jez9999** [duckdb/duckdb#15094] [Ideas]
I'm going to be building an app where I use FTS for searching keywords and titles.  I understand that DuckDB is primarily focused on ingesting lots of info and then parsing it in one go but it'd still be nice if it supported this kind of use case.  The FTS index really needs to be updated incrementally as the data is added/modified/deleted; obviously recreating the whole index each time this happens is going to be hugely wasteful of CPU time.

**Comment by hpvd** [comment]
you are not the only one interested in this:
https://github.com/duckdb/duckdb/discussions/5352
